distributed/deploy/tests/test_adaptive_core.py

```diff
@@ -89,3 +92,21 @@ async def test_interval():
     adapt._target = 10
     await asyncio.sleep(0.020)
     assert len(adapt.plan) == 1  # last value from before, unchanged
+
+
+@pytest.mark.asyncio
+@mock.patch("distributed.deploy.adaptive_core.AdaptiveCore.safe_target")
+async def test_adapt_oserror(mock_safe_target):
+    with tmpfile() as fn:
+        fh = logging.FileHandler(fn)
+        logger = logging.getLogger("distributed.deploy.adaptive_core")
+        logger.addHandler(fh)
+        mock_safe_target.side_effect = OSError
+        adapt = MyAdaptive(minimum=1, maximum=4, wait_count=2)
+        await adapt.adapt()
+        fh.flush()
+        with open(fn) as f:
+            text = f.read()
+            assert "Adaptive stopping due to error" in text
+            assert "Adaptive stop" in text
+        assert not adapt._adapting
```

Comment:
For some reason, the tests are failing sporadically since there is no log msg observered. I'm not sure if this is connected to the filehandler or not but we do have the util function `captured_logger` which registers a streamhandler as the only logger and offers the stream buffer as the ctx mgr return value. This is how we write most tests which are sensitive to logging

```suggestion
    with captured_logger("distributed.deploy.adaptive_core") as log:
        mock_safe_target.side_effect = OSError
        adapt = MyAdaptive(minimum=1, maximum=4)
        await adapt.adapt()
    text = log.getvalue()
    assert "Adaptive stopping due to error" in text
    assert "Adaptive stop" in text
    assert not adapt._adapting
    assert not adapt.periodic_callback
```

Reply:
Thank you, @fjetter